cns/wireserver/net.go

```diff
@@ -49,6 +54,49 @@ func GetPrimaryInterfaceFromResult(res *GetInterfacesResult) (*InterfaceInfo, er
 	return nil, ErrNoPrimaryInterface
 }
 
+// Gets secondary interface details for swiftv2 secondary nics scenario
+func GetSecondaryInterfaceFromResult(res *GetInterfacesResult, macAddress string) (*InterfaceInfo, error) {
+	for _, i := range res.Interface {
+		// skip if primary
+		if i.IsPrimary {
+			continue
+		}
+
+		// skip if no subnets
+		if len(i.IPSubnet) == 0 {
+			continue
+		}
+
+		logger.Printf("i.MacAddress is %s", strings.Split(i.MacAddress, ":"))
+		newMacAddress := regexp.MustCompile(`[^a-zA-Z0-9 ]+`).ReplaceAllString(macAddress, "")
```

Comment:
let's use a better way to handle the newMacAddress

Reply:
added a separate function below for this